Source/loadsave.cpp

```diff
@@ -363,11 +363,13 @@ void LoadPlayer(LoadHelper &file, Player &player)
 	player._pSpellFlags = file.NextLE<uint8_t>();
 	file.Skip(3); // Alignment
 
-	for (auto &spell : player._pSplHotKey)
-		spell = static_cast<spell_id>(file.NextLE<int32_t>());
-
-	for (auto &spellType : player._pSplTHotKey)
-		spellType = static_cast<spell_type>(file.NextLE<int8_t>());
+	// Extra hotkeys: to keep single player save compatibility, read only 4 hotkeys here, rely on LoadHotkeys for the rest
+	for (uint8_t i = 0; i < 4; i++) {

```

Comment:
Thanks for all the great feedback! I haven't done much C since the late 90s on DOS, so I don't know much about all the "new" stuff and the standard library, this is really helpful.

About the constant for the size, I thought about adding one, but I didn't find any place in the code that looked like a central repository of constants that is automatically included in other files. Where would be the best place to add it?

Thanks!

Reply:
We have migrated a few things to C++ as well so that is probably also going to be different from what you are used to :)

Reply:
> About the constant for the size, I thought about adding one, but I didn't find any place in the code that looked like a central repository of constants that is automatically included in other files. Where would be the best place to add it?

In this case I think it would be appropriate to define a `static constexpr` variable inside the player class (in player.h). Something like:

```cpp
// ...
	static constexpr size_t HotKeyCount = 16;
	spell_id _pSplHotKey[Player::HotKeyCount];
	spell_type _pSplTHotKey[Player::HotKeyCount];
// ...
```

I don't know if GCC6 will allow scoping the declaration like this though, it doesn't handle constexpr members in classes particularly well. If that fails to compile moving the declaration with the other constants defined at the top of this file would be fine.